_rules/visible-label-in-accessible-name-2ee8b8.md

```diff
@@ -106,43 +106,52 @@ Mathematical symbols cannot be substituted for text as [explicitly mentioned in
 
 #### Inapplicable Example 1
 
-Not a widget role.
+This `nav` is not a widget, so the [visible][] text does not need to match the [accessible name][].
 
 ```html
-<a aria-label="OK">Next</a>
+<nav aria-label="main nav">W3C navigation</nav>
 ```
 
 #### Inapplicable Example 2
 
-Widget role that does not support name from content.
+This text field does not need to have its [visible][] text match the [accessible name][]. The content of a textfield shows its value instead of its label. The label is usually adjecent to the textfield instead.
 
 ```html
+<div>E-mail</div>
 <input type="email" aria-label="E-mail" value="Contact" />
 ```
 
 #### Inapplicable Example 3
 
-Non-widget role that supports name from content.
+This div has does not have a widget role, so the [visible][] text does not need to match the [accessible name][].
 
 ```html
 <div role="tooltip" aria-label="OK">Next</div>
 ```
 
 #### Inapplicable Example 4
 
-No [rendered text](#rendered-text) in name from content.
+This link has no [visible text content][].
 
 ```html
-<div role="tooltip" aria-label="OK"></div>
+<a href="//w3.org" aria-label="W3C homepage">
+  <img src="/test-assets/shared/w3c-logo.png" alt="w3c logo">
+</a>
 ```
 
 #### Inapplicable Example 5
 
-Non-text content.
+This link has text content of the link is [non-text content][].
```

Comment:
That sentence makes no sense…
1. it is grammatically incorrect
2. there is a high risk of confusion between "text content" and "non-text content". I agree that this is technically correct (the "text content" (HTML sense) of this is "X" which, (according to WCAG) is "non-text content").

Maybe
```suggestion
The content of this link is [non-text content][].
```